packages/flutter/lib/src/widgets/framework.dart

```diff
@@ -4036,13 +4036,11 @@ abstract class Element extends DiagnosticableTree implements BuildContext {
     assert(slots == null || newWidgets.length == slots.length);
 
     Element? replaceWithNullIfForgotten(Element child) {
-      return forgottenChildren != null && forgottenChildren.contains(child) ? null : child;
+      return (forgottenChildren?.contains(child) ?? false) ? null : child;
```

Comment:
What about
```suggestion
      return (forgottenChildren?.contains(child) == true) ? null : child;
```

I'm slightly leaning towards this version but not strong.

Reply:
I had it this way because of the [use_if_null_to_convert_nulls_to_bools](https://dart.dev/lints/use_if_null_to_convert_nulls_to_bools) rule.

Reply:
Ok, that makes sense. :)